x/ibc/07-tendermint/types/evidence.go

```diff
@@ -85,6 +86,17 @@ func (ev Evidence) GetTime() time.Time {
 
 // ValidateBasic implements Evidence interface
 func (ev Evidence) ValidateBasic() error {
+	if ev.Header1.TrustedHeight != ev.Header2.TrustedHeight {
```

Comment:
I personally think this check can be removed, and we allow Headers with different trusted heights. But we would then have to load two trusted consensus states for misbehaviour (one for each header)

Reply:
what would be the benefit of removing this check?

Reply:
Hmm, I see - yes I agree, it should still count as misbehaviour since we're allowing updates from past heights in `updateClient` now

Reply:
I think we should just allow client misbehaviour handlers to load consensus states themselves like we will do in the verification functions

Reply:
should this check be replaced with ensuring the trusted height != 0?

Reply:
I looked into implementing this, it complicates misbehaviour handling fairly significantly. I'll do this in a separate PR since it is a non-breaking change. This can be merged as is assuming there are no additional requests